doc/dev/backend-plugins.md

```diff
@@ -0,0 +1,271 @@
+# Proposal: Backend Plugins
+
+## Backend Contract
+
+There exists a _backend contract_ between `libelektra-kdb` and any plugin acting as a backend plugin.
+This contract sets, the order of the phases described above and defines the interaction between a backend plugin and `libelektra-kdb`.
+
+TODO: update get diagram; add cachecheck phase
+
+![Sequence of phases in `get` operation](kdbGet.svg)
+
+![Sequence of phases in `set` operation](kdbSet.svg)
+
+The diagrams above show the possible sequences of phases during a `get` and a `set` operation.
+For each of the phases of a `get` operation `libelektra-kdb` calls the backend plugin's `elektra<Plugin>Get` function once.
+Similarly, for the phases of a `set` operation `elektra<Plugin>Set` is called.
+
+The current phase is communicated to the backend plugin via the global keyset.
+The value of the key `system:/elektra/kdb/backend/phase` is always set to the current phase.
+
+### Operation `get`
```

Comment:
Do we want a separated Operation `contract`? Or directly return the contract from the plugin factory? This could be the only mandatory function.

Reply:
I think this is a separate issue. How the contract is returned doesn't have anything to do with backend plugins specifically. It is a more general issue that concerns all plugins.

That said, I think separating the contract from `kdbGet` would be a good idea.